channel.go

```diff
@@ -121,6 +136,44 @@ func (c *C[T]) ReadChannel() <-chan T {
 	return c.c
 }
 
+// Rest reads all the values in the channel until it closes, and return them all at once.
+func (c *C[T]) Rest() ([]T, error) {
+	var res []T
+	for {
+		v, err := c.Read()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return nil, err
```

Comment:
```suggestion
			return res, err
```
a consumer maybe care about partial data ?